javascript/javascript3/week2/fetchingAndWaiting.js

```diff
@@ -0,0 +1,17 @@
+function fetchData() {
+    return new Promise(resolve => {
+        fetch('https://yesno.wtf/api/')
+        .then(response => {
+            return response.json();
+        })
+        .then(data => {
+            return data;
+        }).then(data => {
+            setTimeout(() => {
+                resolve(data);
+            }, 3000)
+        })
+    })
+}
+
+fetchData().then(data => console.log(data));
```

Comment:
nice one 👍

Reply:
My opinion is that this code (while it works and is clever) is needlessly complicated. You could write the whole thing without creating a new promise, and just doing everything as one .then chain.